lib/prelude.dx

```diff
@@ -559,8 +591,8 @@ def scan' (init:a) (body:n->a->a) : n=>a = snd $ scan init \i x. dup (body i x)
 def fsum (xs:n=>Float) : Float = yieldAccum \ref. for i. ref += xs i
 def sum  [Add v] (xs:n=>v) : v = reduce zero (+) xs
 def prod [Mul v] (xs:n=>v) : v = reduce one  (*) xs
-def mean (xs:n=>Float) : Float = sum xs / IToF (size n)
-def std (xs:n=>Float) : Float = sqrt $ mean (map sq xs) - sq (mean xs)
+def mean [VSpace v] (xs:n=>v) : v = sum xs / IToF (size n)
+def std [Mul v, VSpace v, Floating v] (xs:n=>v) : v = sqrt $ mean (map sq xs) - sq (mean xs)
```

Comment:
Curious how the extra typeclass indirection affects performance, versus the original functions specialized for `n=>Float`. Would be nice to retroactively benchmark when benchmarking infra is ready.

Reply:
Run-time performance should be identical, because all type classes get fully specialized. There's a small compile-time cost.

Reply:
Awesome, thanks!